### ISSUE TITLE
Investigate transfer of Suggestion DB to IDE as binary not as JSON

### ISSUE THREAD

**Issue by JaroslavTulach** [enso-org/enso#5570]
### Summary

Would it be more effective to transfer suggestion DB (tracked as #5068) & co. as binary, not as JSON? That's a question asked by [startup performance measurements as of Feb 6, 2023](https://docs.google.com/document/d/1-jG7HQoPY7KASG8NZ1PUDzqqd5ZL7wMzVSSMCcNFWMI). 

### Value

Can we speed things up by not sending JSON, but binary data? For example the suggestion DB (tracked as #5068) could be sent as binary. The drawback is that the IDE would have to be modified to accept the protocol change. Sticking to JSON allows us to modify just the engine part.

### Specification

Prototype sending the suggestion DB as binary, not as JSON.

### Acceptance Criteria & Test Cases

- [ ] Store the precomputed Suggestion DB (as done by #5068) in binary form
- [ ] Send the binary Suggestion DB to the IDE in binary form
- [ ] Evaluate the result and decide the next steps

### Related

- #5741 - right now the `Suggestion` list is encoded into binary JSON, sent from engine to `runner.jar`, deserialized, converted to other data structure, serialized into JSON and then sent to the IDE. Unless the multiple serialization/deserialization is eliminated it makes little sense to think about changing the form of transfer.

**Comment by JaroslavTulach** [comment]
Working on #5698 I find out that the transfer of suggestion DB is actually happening in binary channel:
- the runtime messages are [using CBOR factory](https://github.com/enso-org/enso/blob/c965ad34559471071b8f1425f88de9e8be8c547f/engine/polyglot-api/src/main/scala/org/enso/polyglot/runtime/Runtime.scala#L1654)
- suggestions are sent as binary messages
- as [concise binary object representation](https://cbor.io/) indicates the data are converted into binary

Debugging confirms the expectation. The data seem to be sent as binary.

**Comment by farmaazon** [comment]
I see many messages sent through json-rpc connection: 
![image](https://user-images.githubusercontent.com/3919101/220324535-e135c384-e3f9-4413-a871-84a6e4fde9a8.png)
I can easily read them, so they probably are not binary. Or is it some web socket magic?

Are you sure you're not confusing suggestion db updates with visualization value updates? AFAIK, only the latter are sent through binary channel.

**Comment by JaroslavTulach** [comment]
Thanks @farmaazon - there is double serialization/deserialization inside of single(!) LS/engine process - maybe I got confused by that.

**Comment by wdanilo** [comment]
@JaroslavTulach I fully agree. All data should be in a binary format, minimized, and as small as possible.
